echidna/core/fill_spectrum.py

```diff
@@ -0,0 +1,87 @@
+import ROOT
+import rat
+import math
+import echidna.core.spectra as spectra
+
+def _scint_weights(times, T):
+    """This method applies to the scintillator backgrounds.
+    It produces the list of weights relative to each time period. 
+    The calculation of weights is based on radioactive decay formula.
+ 
+    Args:
+      times (*list* of *int*): Time periods 
+      T (float): The Half-life of a studied background
+
+    Returns:
+      Weights (*list* of *float*)
+    """
+    weights = [0]
+    for time in times:
+        weights.append( math.exp(-time/T) )
+    return (weights)
+
+def _av_weights(times, T):
+    """This method applies to the backgrounds due to AV leaching.
+    It produces the list of weights relative to each time period. 
+    The calculation of weights is based on radioactive decay formula.
+ 
+    Args:
+      times (*list* of *int*): Time periods 
+      T (float): The Half-life of a studied background
+
+    Returns:
+      Weights (*list* of *float*)
+    """
+    weights = [0]
+    for time in times:
+        weights.append( 1.0 )
+    return (weights)
+
+def fill_spectrum(filename, spectrumname, T):
+    """This function fills in the ndarray of energies, radii, times 
+    and weights. It takes the reconstructed energies and positions
+    of the events from the root file. In order to keep the statistics, 
+    the time dependence is performed via adding weights to every event
+    depending on the time period. Both, studied time and Half-life must 
+    be written in the same units.  
+    
+    Args:
+      filename (str): A root file to study 
+      spectrumname (str): A name of future ndarray 
+      T (float): The Half-life of a studied background
+
+      Returns:
+        spectrum (spectra.Spectra) 
```

Comment:
If you format this as 

    Returns:
      spectrum (:class:`echidna.core.spectra.Spectra`)

sphinx should be able to understand the type.